src/main/java/com/maxmind/db/SingleBuffer.java

```diff
@@ -0,0 +1,174 @@
+package com.maxmind.db;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.channels.FileChannel.MapMode;
+import java.nio.charset.CharacterCodingException;
+import java.nio.charset.CharsetDecoder;
+
+/**
+ * A {@link Buffer} implementation backed by a single {@link ByteBuffer}.
+ *
+ * <p>This implementation is limited to capacities up to
+ * {@link Integer#MAX_VALUE}, as {@link ByteBuffer} cannot exceed that size.
+ */
+public class SingleBuffer implements Buffer {
```

Comment:
We don't want this to be part of the public API. It and its methods can be package-private.